demo_nodes_cpp/src/parameters/even_parameters_node.cpp

```diff
@@ -70,6 +73,7 @@ class EvenParameterNode : public rclcpp::Node
               "requested value for parameter '%s' is not an even number, rejecting change...",
               parameter.get_name().c_str()
             );
+            result.reason = "only integer parameters can be set";
```

Comment:
Can we actually get to this case now that parameter types are static and enforced by rclcpp?

Reply:
We can in this particular case because this node has `allow_undeclared_parameters` set to true.  Thus it acts like a parameter blackboard.  Here's a session from me testing this out:

```
$ ros2 param set /even_parameters_node myint 2
Set parameter successful
$ ros2 param set /even_parameters_node myint 1
Setting parameter failed: only even integers can be set
$ ros2 param set /even_parameters_node mydouble 1.0
Setting parameter failed: only integer parameters can be set
```